contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,309 @@
+	// that are allowed to connect to the target
+	DesiredReplicationFactor int `json:"desiredReplicationFactor"`
+	//ReplicaDetails refers to the trusty replica information
+	ReplicaDetails CStorVolumeReplicaDetails `json:"replicaDetails,omitempty"`
+}
+
+// ReplicaID is to hold replicaID information
+type ReplicaID string
+
+// CStorVolumePhase is to hold result of action.
+type CStorVolumePhase string
+
+// CStorVolumeStatus is for handling status of cvr.
+type CStorVolumeStatus struct {
+	Phase           CStorVolumePhase `json:"phase"`
+	ReplicaStatuses []ReplicaStatus  `json:"replicaStatuses,omitempty"`
+	// Represents the actual resources of the underlying volume.
+	Capacity resource.Quantity `json:"capacity,omitempty"`
+	// LastTransitionTime refers to the time when the phase changes
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	LastUpdateTime     metav1.Time `json:"lastUpdateTime,omitempty"`
+	Message            string      `json:"message,omitempty"`
+	// Current Condition of cstorvolume. If underlying persistent volume is being
+	// resized then the Condition will be set to 'ResizePending'.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []CStorVolumeCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,4,rep,name=conditions"`
+	// ReplicaDetails refers to the trusty replica information which are
+	// connected at given time
+	ReplicaDetails CStorVolumeReplicaDetails `json:"replicaDetails,omitempty"`
+}
+
+// CStorVolumeReplicaDetails contains trusty replica inform which will be
+// updated by target
+type CStorVolumeReplicaDetails struct {
+	// KnownReplicas represents the replicas that target can trust to read data
+	KnownReplicas map[ReplicaID]string `json:"knownReplicas,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CStorVolumeList is a list of CStorVolume resources
+type CStorVolumeList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorVolume `json:"items"`
+}
+
+// CVStatusResponse stores the reponse of istgt replica command output
+// It may contain several volumes
+type CVStatusResponse struct {
+	CVStatuses []CVStatus `json:"volumeStatus"`
+}
+
+// CVStatus stores the status of a CstorVolume obtained from response
+type CVStatus struct {
+	Name            string          `json:"name"`
+	Status          string          `json:"status"`
+	ReplicaStatuses []ReplicaStatus `json:"replicaStatus"`
+}
+
+// ReplicaStatus stores the status of replicas
+type ReplicaStatus struct {
+	ID                string `json:"replicaId"`
+	Mode              string `json:"mode"`
+	CheckpointedIOSeq string `json:"checkpointedIOSeq"`
+	InflightRead      string `json:"inflightRead"`
+	InflightWrite     string `json:"inflightWrite"`
+	InflightSync      string `json:"inflightSync"`
+	UpTime            int    `json:"upTime"`
+	Quorum            string `json:"quorum"`
```

Comment:
Q: What does Quorum mean or indicate? Similarly, what is CheckPointedIOSeq?

Reply:
Sure, update the information related to all the fields.